app/controllers/sections/items/ItemPackagingSealChoiceController.scala

```diff
@@ -47,20 +47,24 @@ class ItemPackagingSealChoiceController @Inject()(
   def onPageLoad(ern: String, draftId: String, itemIdx: Index, packagingIdx: Index, mode: Mode): Action[AnyContent] =
     authorisedDataRequestAsync(ern, draftId) { implicit request =>
       validatePackagingIndexAsync(itemIdx, packagingIdx) {
-        withItemPackaging(itemIdx, packagingIdx) { description =>
-          renderView(Ok, fillForm(ItemPackagingSealChoicePage(itemIdx, packagingIdx), formProvider()), itemIdx, packagingIdx, mode, description)
+        withItemPackagingQuantity(itemIdx, packagingIdx) { packagingQuantity =>
+          withItemPackaging(itemIdx, packagingIdx) { description =>
+            renderView(Ok, fillForm(ItemPackagingSealChoicePage(itemIdx, packagingIdx), formProvider()), itemIdx, packagingIdx, packagingQuantity, mode, description)
+          }
         }
       }
     }
 
   def onSubmit(ern: String, draftId: String, itemIdx: Index, packagingIdx: Index, mode: Mode): Action[AnyContent] =
     authorisedDataRequestAsync(ern, draftId) { implicit request =>
       validatePackagingIndexAsync(itemIdx, packagingIdx) {
-        withItemPackaging(itemIdx, packagingIdx) { description =>
-          formProvider().bindFromRequest().fold(
-            renderView(BadRequest, _, itemIdx, packagingIdx, mode, description),
-            cleanseSaveAndRedirect(_, itemIdx, packagingIdx, mode)
-          )
+        withItemPackagingQuantity(itemIdx, packagingIdx) { packagingQuantity =>
```

Comment:
What do you think to moving `withItemPackagingQuantity` into the render view method instead?